TopMovies/MoviesConnector/CategoryVCConnector.swift

```diff
@@ -0,0 +1,122 @@
+      _updateProps(props)
+      isExistingMoviesConnected = !isExistingMoviesConnected
+      downloadedMovies = moviesList
+    }
+    let categoryState = state.categoriesPaginationState.paginated[categoryId]
+    if isDownloadingInProgress(categoryState)
+    {
+      var isReloadInProgress = false
+      var isLoadMoreInProgress = false
+      if case .downloading = categoryState?.reload {
+        isReloadInProgress = true
+      }
+      if case .downloading = categoryState?.loadMore {
+        isLoadMoreInProgress = true
+      }
+      let props = MoviesCategoryVCProps(
+        categoryId: categoryId,
+        categoryName: categoryName,
+        loadProgress: .init(isReloadInProgress: isReloadInProgress,
+                            isLoadMoreInProgress: isLoadMoreInProgress),
+        moviesState: .init(hasNewData: false,
+                           moviesUsage: .reload,
+                           movies: []))
+      _updateProps(props)
+    } else {
+      switch moviesList.count {
+      case (downloadedMovies.count + 1)...:
+        var listDiff = moviesList
+        listDiff.removeSubrange(0..<downloadedMovies.count)
+        let props = MoviesCategoryVCProps(
+          categoryId: categoryId,
+          categoryName: categoryName,
+          loadProgress: .init(isReloadInProgress: false,
+                              isLoadMoreInProgress: false),
+          moviesState: .init(hasNewData: true,
+                             moviesUsage: .loadMore,
+                             movies: listDiff
+                              .compactMap {
+                                MovieTableViewCellProps(movie: state.moviesState.relational[$0])
+                              }))
+        _updateProps(props)
+        downloadedMovies += listDiff
+      case ...(downloadedMovies.count) where moviesList != downloadedMovies:
+        let props = MoviesCategoryVCProps(
+          categoryId: categoryId,
+          categoryName: categoryName,
+          loadProgress: .init(isReloadInProgress: false,
+                              isLoadMoreInProgress: false),
+          moviesState: .init(hasNewData: true,
+                             moviesUsage: .reload,
+                             movies: moviesList
+                              .compactMap {
+                                MovieTableViewCellProps(movie: state.moviesState.relational[$0])
+                              }))
+        downloadedMovies = moviesList
+        _updateProps(props)
+      case downloadedMovies.count:
+        let props = MoviesCategoryVCProps(
+          categoryId: categoryId,
+          categoryName: categoryName,
+          loadProgress: .init(isReloadInProgress: false,
+                              isLoadMoreInProgress: false),
+          moviesState: .init(hasNewData: false,
+                             moviesUsage: .reload,
+                             movies: []))
```

Comment:
You create props for like 5 times in that function. Why don't you create them just once and simply calculate it's specific values in a separate function with only required parameters? It will make it easier to read your code and to understand what's going on.